assets/js/worker.js

```diff
@@ -1,18 +1,38 @@
-onmessage = function(event) {
-	self.importScripts('https://unpkg.com/lunr/lunr.js');
-	
+let scriptLoaded = false;
+
+onmessage = async function(event) {
+	const scriptUrl = 'https://unpkg.com/lunr/lunr.js';
+  const integrityHash = 'sha256-lDFybwXA6uKm5U3Bl3CUIoafJcrUTyQw0vt92ugMxxc=';
+	if (scriptLoaded) return
+
+	let response
+	if (!scriptLoaded) {
+		try {
+			response = await fetch(scriptUrl, { integrity: integrityHash });
+		} catch (err) {
+			console.log(err)
+			return
+		}
+	}
+	const scriptText = await response.text();
+
+	// Create a script element and append the loaded script
+	const scriptElement = new Function(scriptText);
+	scriptElement();
```

Comment:
/nit comment and variable name are "lying" 😅 the code is not creating a script element, it's just creating a local function and executing it.

Reply:
By the way if you have strict CSPs, the `Function` constructor (basically `eval()`) might not be allowed...